nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -439,7 +440,31 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
             # after dist_checkpointing.load, sharded tensors will be replaced with tensors
             checkpoint['state_dict'] = sharded_state_dict
             checkpoint['optimizer_states'] = [self.optimizer_sharded_state_dict()]
-            return self.checkpoint_io.load_checkpoint(checkpoint_path, sharded_state_dict=checkpoint)
+            
```

Comment:
can you make this whole block another function? also can you add a docstrings what the (new function) is doing?